pkg/controller/utils/component.go

```diff
@@ -297,6 +302,11 @@ func mergeState(desired client.Object, current runtime.Object) client.Object {
 		dsa.Spec.ElasticsearchRef = csa.Spec.ElasticsearchRef
 		dsa.Status = csa.Status
 		return dsa
+	case *apiextenv1.CustomResourceDefinition:
+		c := current.(*apiextenv1.CustomResourceDefinition)
+		d := desired.(*apiextenv1.CustomResourceDefinition)
+		d.SetResourceVersion(c.GetResourceVersion())
```

Comment:
Is this correct? Would we want the resource version to be incremented, or fail if the resource current resource version is ahead of the desired (this protects against race conditions and reverting changes).

Reply:
> that meant the code above would not replace the ResourceVersion because there was already one set

Why do we want to replace the resource version though? If you were to edit a resource, the best way (I thought) would be to pull the resource with the resource version, edit it, then write it back. That way if something else did this simultaneously then this write would fail because the resource version would be different. I might be missing something though....

Reply:
We want our desired resource to have the "current" ResourceVersion so when we do the update if something else has edited the same resource our update will fail.
The whole point of this mergeState is to merge our 'desired' resource with the 'current' one that was retrieved, this function is doing what you are suggesting we need to do.
I'm not quite sure why we don't just overwrite the desired ResourceVersion with the current ResourceVersion. That should be something we can always do.